src/controllers/ArtsController.js

```diff
@@ -417,6 +419,235 @@ class ArtsController {
       return res.status(response.code).json(response);
     }
   }
+
+  /**
+   * @static
+   * @desc POST /api/v1/arts/:artId/like
+   * @param {object} req
+   * @param {object} res
+   * @memberof ArtsController
+   * @returns {object} successful like
+   */
+  static async likeArticle(req, res) {
+    try {
+      const { artId } = req.params;
+      const { id } = req.verifyUser;
+
+      const like = await Like.findOrCreate({
+        where: {
+          artId,
+          userId: id
+        },
+      })
+        .spread((user, created) => {
+          user.get({ plain: true });
+          return created;
+        });
+
+      if (!like) {
+        const response = new Response(
+          'Bad Request',
+          400,
+          'You have already liked this article',
+        );
+        return res.status(response.code).json(response);
+      }
+
+      LikeUnlike.like(artId);
+
+      const checkIfDisliked = await Like.findOne({
+        where: {
+          artId,
+          userId: id
+        },
+      });
+
+      if (checkIfDisliked) {
+        await Dislike.destroy({
+          where: {
+            artId,
+            userId: id
+          },
+        });
+        DisikeUndislike.undislike(artId);
+      }
+
+      const response = new Response(
+        'Ok',
+        201,
+        `You just liked article ${artId}`,
+      );
+      return res.status(response.code).json(response);
+    } catch (err) {
+      const response = new Response(
+        'Internal server error',
+        500,
+        `${err}`,
+      );
+      return res.status(response.code).json(response);
+    }
+  }
+
+  /**
+  * @static
+  * @desc POST /api/v1/arts/:artId/unlike
+  * @param {object} req
+  * @param {object} res
+  * @memberof ArtsController
+  * @returns {object} successful unlike
+  */
+  static async unlikeArticle(req, res) {
```

Comment:
I think you should be able to handle like and unlike in one controller.

Reply:
Yeah, I know it's possible. I thought that too but I felt separate controllers could work as well.

But if you suggest I have them in one controller, it's fine

Reply:
Yea. One controller.